### PR TITLE
Add address field to memcached plugin

### PR DESCRIPTION
**Pull request by mrunge**
As addition to the host field, there is also the address setting,
which is helpful sometimes.

#### Pull Request (PR) description
This also adds the address field to the memcached plugin.

### CODE REVIEW

File: templates/plugin/memcached.conf.erb [hunk 1]
```diff
@@ -1,9 +1,16 @@
 <Plugin memcached>
```
Comment (bastelfreak) [inline templates/plugin/memcached.conf.erb]: hey @mrunge, are you interested in turning this into an epp template?

File: manifests/plugin/memcached.pp [hunk 2]
```diff
@@ -1,7 +1,10 @@
 # https://collectd.org/wiki/index.php/Plugin:memcached
 class collectd::plugin::memcached (
   $ensure         = 'present',
-  Hash $instances = {'default' => {'host' => '127.0.0.1', 'port' => 11211 } },
+  Hash $instances = {'default' => {
```
Comment (bastelfreak) [inline manifests/plugin/memcached.pp]: Can you please enforce the structure of the hash?

Comment (mrunge) [reply manifests/plugin/memcached.pp]: What do you mean by 'enforcing the structure' here? I'm not sure that I can follow you.

**Comment by mrunge** [comment]
Ok, now the tests report a valid issue, I'll submit an update asap.

**Comment by mrunge** [comment]
I see the tests failing, but not connected to the change.

On my setup, tests passed, and on 5 of the CI runners;

**Comment by mrunge** [comment]
Ok, the error has been
```
# ArgumentError:
#   wrong number of arguments (given 0, expected 2)
#   ./vendor/bundle/ruby/2.5.0/gems/beaker-puppet-1.3.0/lib/beaker-puppet/helpers/host_helpers.rb:17:in 'stat'
```

**Comment by mrunge** [comment]
I'm still not sure what's happening in the tests. It seems not connected.

**Comment by bastelfreak** [comment]
I restarted the tests here, it should now work.

**Comment by bastelfreak** [comment]
This is another bug in beaker-puppet/beaker :(

**Comment by mrunge** [comment]
Any ideas or how to continue?

**Comment by bastelfreak** [comment]
@alexjfisher do you know if there was any progress on the beaker issue? I lost track.

**Comment by alexjfisher** [comment]
I think https://github.com/puppetlabs/beaker-puppet/pull/59 might have caused a regression?

**Comment by alexjfisher** [comment]
But I can't see how or why yet.  I'll do some more testing.

**Comment by alexjfisher** [comment]
beaker failures fixed in https://github.com/voxpupuli/puppet-collectd/pull/858

**Comment by bastelfreak** [comment]
@mrunge can you rebase please?

**Comment by mrunge** [comment]
Is anything missing here from my side?

File: templates/plugin/memcached.conf.erb (hunk 1 above)
Comment (juniorsysadmin) [reply templates/plugin/memcached.conf.erb]: @mrunge We prefer epp for new templates going forward, but will merge this for now.

**Comment by mrunge** [comment]
Thank you.

**Pull request merged**